src/composables/transactionData.ts

```diff
@@ -46,8 +48,8 @@ import { useAeSdk } from './aeSdk';
 import { useTippingContracts } from './tippingContracts';
 
 interface UseTransactionOptions {
-  transaction?: ITransaction;
-  externalAddress?: AccountAddress;
+  transaction: Ref<ITransaction | undefined>;
+  transactionCustomOwner?: Ref<AccountAddress | undefined>;
```

Comment:
These are the main changes @CedrikNikita @martinkaintas 
Now instead of using `setTransaction` we just pass the transaction reference and allow to be modified in the scope of component.